src/main/java/stef/projects/console/repository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,170 @@
+        }
+        return user;
+    }
+
+    @Override
+    public List<User> findUsersByUserRole(UserRole userRole) throws SQLException {
+        PreparedStatement statement = DatabaseConnection.getPreparedStatementFromQuery(SELECT_BY_USER_ROLE_QUERY);
+        statement.setString(1, UserRole.getValue(userRole));
+        ResultSet resultSet = statement.executeQuery();
+        List<User> userList = new ArrayList<>();
+        while (resultSet.next()) {
+            showUser(resultSet);
+            User user = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+            userList.add(user);
+        }
+        return userList;
+    }
+
+    @Override
+    public List<User> findUserByFirstNameOrLastName(String fistName, String lastName) throws SQLException {
+        PreparedStatement statement = DatabaseConnection.getPreparedStatementFromQuery(SELECT_BY_FIRST_OR_LAST_NAME_QUERY);
+        statement.setString(1, fistName);
+        statement.setString(2, lastName);
+        ResultSet resultSet = statement.executeQuery();
+        List<User> userList = new ArrayList<>();
+        while (resultSet.next()) {
+            showUser(resultSet);
+            User user = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+            userList.add(user);
+        }
+        return userList;
+    }
+
+    private boolean extractStatus(int status) {
+        if (status == 1) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private void fillPreparedStatement(User user, PreparedStatement statement, boolean shouldUpdate) throws SQLException {
+        statement.setString(1, user.getFirstName());
+        statement.setString(2, user.getLastName());
+        statement.setString(3, user.getUsername());
+        statement.setString(4, user.getEmail());
+        statement.setString(5, user.getPassword());
+        statement.setLong(6, UserRole.getNumberValue(user.getUserRole()));
+        if (shouldUpdate) {
+            statement.setLong(7, user.getId());
+        }
+    }
+
+    private void showUser(ResultSet resultSet) throws SQLException {
+        System.out.print(resultSet.getLong("id") + ", ");
+        System.out.print(resultSet.getString("first_name") + ", ");
+        System.out.print(resultSet.getString("last_name") + ", ");
+        System.out.print(resultSet.getString("username") + ", ");
+        System.out.print(resultSet.getString("email") + ", ");
+        System.out.print(resultSet.getString("password") + " ");
+        System.out.println(resultSet.getString("value") + " ");
+    }
+
```

Comment:
Make this method accept a ResultSet instead and do all the .getStuff() there in that method